api/objects.proto

```diff
@@ -295,6 +293,20 @@ message Secret {
 	bool internal = 4;
 }
 
+// Config represents a set of configuration files that should be passed to a
+// container.
+message Config {
+	option (docker.protobuf.plugin.store_object) = { };
+
+	string id = 1;
+
+	Meta meta = 2 [(gogoproto.nullable) = false];
+
+	// Spec contains the actual config data, as well as any context around the
+	// config data that the user provides.
+	ConfigSpec spec = 3  [(gogoproto.nullable) = false];
```

Comment:
Do you foresee any use cases where we would want to treat something as an internal config rather than an internal secret? I guess the main differences between secrets and configs are:

- We redact payload at the API endpoint - but this shouldn't matter for internal secrets/configs, because they're internal
- Secrets are mounted to a fixed position - but this doesn't apply for internal secrets.

I was thinking we should keep it simple and not add internal configs just to mirror secrets, but if this is a useful feature and/or necessary for forward compatibility I can add it back.